src/item.cpp

```diff
@@ -1965,14 +1961,40 @@ nc_color item::color_in_inventory() const
             case NO_TOOL:
                 break;
         }
-    } else if (is_ammo()) { // Likewise, ammo is green if you have guns that use it
+    } else if( is_gun() ) {
+        // Guns are green if you are carrying ammo for them
+        // Yellow if you have ammo but no mags
+        // Gun with integrated mag counts as both
         ammotype amtype = ammo_type();
-        if (u->weapon.is_gun() && u->weapon.ammo_type() == amtype) {
+        bool has_ammo = u->has_ammo( amtype );
+        bool has_mag = u->has_magazine_for_gun( *this );
+        if( has_ammo && has_mag ) {
             ret = c_green;
-        } else {
-            if (u->has_gun_for_ammo(amtype)) {
-                ret = c_green;
-            }
+        } else if( has_ammo || has_mag ) {
+            ret = c_yellow;
```

Comment:
Do active items not use `c_yellow`? If so it might be best to select another colour or just show the result as `c_red`/`c_ltred`